From: Prarit Bhargava <prarit@redhat.com>
Date: Wed, 15 Oct 2008 06:35:33 -0400
Subject: [agp] re-introduce 82G965 graphics support
Message-id: 20081015103533.4034.660.sendpatchset@prarit.bos.redhat.com
O-Subject: [RHEL5.3 PATCH] Re-introduce 82G965 graphics support
Bugzilla: 466307
RH-Acked-by: Peter Martuccelli <peterm@redhat.com>
RH-Acked-by: Brian Maly <bmaly@redhat.com>
RH-Acked-by: Ivan Vecera <ivecera@redhat.com>
RH-Acked-by: Dave Airlie <airlied@redhat.com>
RH-Acked-by: Mauro Carvalho Chehab <mchehab@redhat.com>

Re-introduce 82G965 graphics support into intel-agp driver.  Code was
accidentally removed in patch:

b05cb1ebfe78ee105014e7a6c2a7ad38573d91a8

Successfully compiled by me, and tested by jturner on his system.

Resolves BZ 466307.

diff --git a/drivers/char/agp/intel-agp.c b/drivers/char/agp/intel-agp.c
index 3e627b6..f72fe60 100644
--- a/drivers/char/agp/intel-agp.c
+++ b/drivers/char/agp/intel-agp.c
@@ -2013,6 +2013,13 @@ static int __devinit agp_intel_probe(struct pci_dev *pdev,
 		} else
 			bridge->driver = NULL;
 		break;
+	case PCI_DEVICE_ID_INTEL_82965G_HB:
+		if (find_i830(PCI_DEVICE_ID_INTEL_82965G_IG)) {
+			bridge->driver = &intel_i965_driver;
+			name = "965G";
+		} else
+			bridge->driver = NULL;
+		break;
 	case PCI_DEVICE_ID_INTEL_82965GM_HB:
 		if (find_i830(PCI_DEVICE_ID_INTEL_82965GM_IG)) {
 			bridge->driver = &intel_i965_driver;
